core/tests/puppeteer-acceptance-tests/puppeteer-testing-utilities/puppeteer-utils.ts

```diff
@@ -173,6 +173,31 @@ export class BaseUser {
     inputUploadHandle.uploadFile(fileToUpload);
   }
 
+  /**
+   * This function validates whether an anchor tag is correctly linked
+   * to external PDFs or not. Use this particularly when interacting with
+   * buttons associated with external PDF links, because Puppeteer in
+   * headless mode, does not natively support opening of external PDFs.
+   */
+  async openExternalPdfLink(
+    selector: string,
+    expectedUrl: string
+  ): Promise<void> {
+    await this.page.waitForSelector(selector);
+    const href = await this.page.$eval(selector, element =>
+      element.getAttribute('href')
+    );
+    if (href === null) {
+      throw new Error(`The ${selector} is not an anchor tag!`);
+    }
+    if (href !== expectedUrl) {
+      throw new Error(
+        `The ${selector} does not open the expected URL!
```

Comment:
Can you demonstrate what this error message looks like when printed (when interpolated with the name of the selector)? Also, what stack trace is shown?

I just want to verify that the error message produced here is easily debuggable by developers.

Reply:
Sure, here's the produced stack trace.

```Failures:
1. Logged-in User in About Foundation page should open the page with the Even Those Who Are In School link.
Message:
Error: Error: The .e2e-test-about-foundation-page-millions-of-content a:nth-child(2) does not open the expected URL!
         Actual: http://uis.unesco.org/sites/default/files/documents/fs46-more-than-half-children-not-learning-en-2017.pdf, Expected: https://uis.unesco.org/sites/default/files/documents/fs46-more-than-half-children-not-learning-en-2017.pdf
Stack:
Error: Error: The .e2e-test-about-foundation-page-millions-of-content a:nth-child(2) does not open the expected URL!
         Actual: http://uis.unesco.org/sites/default/files/documents/fs46-more-than-half-children-not-learning-en-2017.pdf, Expected: https://uis.unesco.org/sites/default/files/documents/fs46-more-than-half-children-not-learning-en-2017.pdf
    at BaseUser.clickEvenThoseWhoAreInSchoolLinkInAboutFoundation (/Users/akhilesh/opensource/oppia/core/tests/puppeteer-acceptance-tests/user-utilities/logged-in-users-utils.ts:342:13)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at UserContext.<anonymous> (/Users/akhilesh/opensource/oppia/core/tests/puppeteer-acceptance-tests/spec/logged-in-user-tests/click-all-buttons-in-about-foundation-page.spec.ts:51:7)```

Reply:
I have updated the error message. Please take a look and  let me know if any changes are required.